be/src/exec/olap_scanner.cpp

```diff
@@ -335,6 +335,7 @@ Status OlapScanner::get_batch(RuntimeState* state, RowBatch* batch, bool* eof) {
                         // make sure to reset null indicators since we're overwriting
                         // the tuple assembled for the previous row
                         tuple->init(_tuple_desc->byte_size());
+                        batch->agg_object_pool()->remove_last_one();
```

Comment:
Could you give some more detail about the OOM?

If one thread(T2) add an new object into agg_object_pool, just before remove_last_one called by this thread(T1).
I think it may remove object added by T2, not the one added by T1?

Reply:
_batch_ in _Status OlapScanner::get_batch(RuntimeState*, RowBatch*, bool*)_ was allocated by _void OlapScanNode::scanner_thread(OlapScanner*)_, every scanner_thread has its own _batch_, it not a multi-thread case。

Reply:
Thanks for your reply.

1. Could you give me some detail steps? I want to try to reproduce the problem.
2. If one tuple has bitmap and hll column at the same time, remove_last_one should be called twice?